src/de/simple_type.rs

```diff
@@ -0,0 +1,1311 @@
+                        // the initial borrowing
+                        Content::Input(s) => {
+                            let (item, rest) = s.split_at(end);
+                            self.content = Some(Content::Input(rest));
+
+                            seed.deserialize(AtomicDeserializer {
+                                content: Content::Input(item),
+                                escaped: self.escaped,
+                            })
+                        }
+                        Content::Slice(s) => {
+                            let (item, rest) = s.split_at(end);
+                            self.content = Some(Content::Slice(rest));
+
+                            seed.deserialize(AtomicDeserializer {
+                                content: Content::Slice(item),
+                                escaped: self.escaped,
+                            })
+                        }
+                        // Skip additional bytes if we own data for next iteration, but deserialize from
+                        // the borrowed data from our buffer
+                        #[cfg(feature = "encoding")]
+                        Content::Owned(s, skip) => {
+                            let item = s.split_at(skip + end).0;
+                            let result = seed.deserialize(AtomicDeserializer {
+                                content: Content::Slice(item),
+                                escaped: self.escaped,
+                            });
+
+                            self.content = Some(Content::Owned(s, skip + end));
+
+                            result
+                        }
+                    },
+                }
+                .map(Some);
+            }
+        }
+        Ok(None)
+    }
+}
+
+////////////////////////////////////////////////////////////////////////////////////////////////////
+
+/// A version of [`Cow`] that can borrow from two different buffers, one of them
+/// is a deserializer input.
+///
+/// # Lifetimes
+/// - `'de` -- lifetime of the data that deserializer borrow from the parsed input
+/// - `'a` -- lifetime of the data that owned by a deserializer
+enum CowRef<'de, 'a> {
+    /// An input borrowed from the parsed data
+    Input(&'de [u8]),
+    /// An input borrowed from the buffer owned by another deserializer
+    Slice(&'a [u8]),
+    /// An input taken from an external deserializer, owned by that deserializer
+    Owned(Vec<u8>),
+}
+impl<'de, 'a> Deref for CowRef<'de, 'a> {
+    type Target = [u8];
+
+    fn deref(&self) -> &[u8] {
+        match self {
+            Self::Input(slice) => slice,
+            Self::Slice(slice) => slice,
+            Self::Owned(ref v) => v,
+        }
+    }
+}
+
+/// A deserializer for an xml probably escaped and encoded value of XSD [simple types].
+/// This deserializer will borrow from the input as much as possible.
+///
+/// When asking for any date, deserializer always return a whole string that
+/// it contain.
```

Comment:
Not sure what this is saying

Reply:
This is a misprint -- should be "When asking for any data". This sentence should mean, that calling `Deserializer::deserialize_any()` always will call `Visitor::visit_str/visit_borrowed_str/visit_string`, because the string is the only type that XML stores internally (i.e. without type hints you cannot understand what type have, for example, `1` in `<tag>1</tag>`). I agree that this sentence is too non-obvious even when misprint is fixed.

Reply:
Nice explanation, thanks.